sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -54,7 +54,7 @@ object CollationTypeCasts extends TypeCoercionRule {
 
     case otherExpr @ (
       _: In | _: InSubquery | _: CreateArray | _: ArrayJoin | _: Concat | _: Greatest | _: Least |
-      _: Coalesce | _: BinaryExpression | _: ConcatWs) =>
+      _: Coalesce | _: BinaryExpression | _: ConcatWs | _: Mask) =>
```

Comment:
should `StringToMap` have the same requirement?

Reply:
For StringToMap, the second and third parameters are treated as regex:
`Both `pairDelim` and `keyValueDelim` are treated as regular expressions."`

so no casting there, and their collation shouldn't be taken into consideration